test/ICSharpCode.SharpZipLib.Tests/Zip/ZipEncryptionHandling.cs

```diff
@@ -126,40 +127,80 @@ public void ZipFileAesDecryption()
-		[TestCase(0, CompressionMethod.Stored, true)]
-		[TestCase(128, CompressionMethod.Deflated, true)]
-		[TestCase(128, CompressionMethod.Stored, true)]
-		[TestCase(256, CompressionMethod.Deflated, true)]
-		[TestCase(256, CompressionMethod.Stored, true)]
-		public void ZipInputStreamDecryption(int aesKeySize, CompressionMethod compressionMethod, bool forceDataDescriptor)
+		[TestCase(0,   CompressionMethod.Deflated, false, false, false)]
+		[TestCase(0,   CompressionMethod.Stored,   false, false, false)]
+		[TestCase(128, CompressionMethod.Deflated, false, false, false)]
+		[TestCase(128, CompressionMethod.Stored,   false, false, false)]
+		[TestCase(256, CompressionMethod.Deflated, false, false, false)]
+		[TestCase(256, CompressionMethod.Stored,   false, false, false)]
+		[TestCase(0,   CompressionMethod.Deflated, true,  false, false)]
+		[TestCase(0,   CompressionMethod.Stored,   true,  false, false)]
+		[TestCase(128, CompressionMethod.Deflated, true,  false, false)]
+		[TestCase(128, CompressionMethod.Stored,   true,  false, false)]
+		[TestCase(256, CompressionMethod.Deflated, true,  false, false)]
+		[TestCase(256, CompressionMethod.Stored,   true,  false, false)]
+		
+		[TestCase(0,   CompressionMethod.Deflated, false, true,  false)]
+		[TestCase(0,   CompressionMethod.Stored,   false, true,  false)]
+		[TestCase(128, CompressionMethod.Deflated, false, true,  false)]
+		[TestCase(128, CompressionMethod.Stored,   false, true,  false)]
+		[TestCase(256, CompressionMethod.Deflated, false, true,  false)]
+		[TestCase(256, CompressionMethod.Stored,   false, true,  false)]
+		[TestCase(0,   CompressionMethod.Deflated, true,  true,  false)]
+		[TestCase(0,   CompressionMethod.Stored,   true,  true,  false)]
+		[TestCase(128, CompressionMethod.Deflated, true,  true,  false)]
+		[TestCase(128, CompressionMethod.Stored,   true,  true,  false)]
+		[TestCase(256, CompressionMethod.Deflated, true,  true,  false)]
+		[TestCase(256, CompressionMethod.Stored,   true,  true,  false)]
+		
+		[TestCase(0,   CompressionMethod.Deflated, false, false, true)]
+		[TestCase(0,   CompressionMethod.Stored,   false, false, true)]
+		[TestCase(128, CompressionMethod.Deflated, false, false, true)]
+		[TestCase(128, CompressionMethod.Stored,   false, false, true)]
+		[TestCase(256, CompressionMethod.Deflated, false, false, true)]
+		[TestCase(256, CompressionMethod.Stored,   false, false, true)]
+		[TestCase(0,   CompressionMethod.Deflated, true,  false, true)]
+		[TestCase(0,   CompressionMethod.Stored,   true,  false, true)]
+		[TestCase(128, CompressionMethod.Deflated, true,  false, true)]
+		[TestCase(128, CompressionMethod.Stored,   true,  false, true)]
+		[TestCase(256, CompressionMethod.Deflated, true,  false, true)]
+		[TestCase(256, CompressionMethod.Stored,   true,  false, true)]		
+		public void ZipInputStreamDecryption(int aesKeySize, CompressionMethod compressionMethod, bool forceDataDescriptor, bool skipEntries, bool partiallyReadEntries)
```

Comment:
This is getting out hand...
Perhaps use [ValuesAttribute](https://docs.nunit.org/articles/nunit/writing-tests/attributes/values.html) instead?

Reply:
I can change that although it would add the combination skipEntries = true & partiallyReadEntries = true which is not really needed and would therefore result in 12 additional unnecessary test cases.

Reply:
```cs
enum EntryMode { Skip, PartiallyRead, FullRead }
```
It's not readable the way it is right now. And how would you know if you missed a combination?

Reply:
Actually, I think these should be separate tests all together. It seems like you are testing different scenarios.
Just copy the whole test and make changes to it with an updated description and name, the encryption level is probably not necessary to have all variants of, but include it if you disagree.

Reply:
Sepatarted into several tests